EncryptedChannel/EncryptedChannelServer.cs

```diff
@@ -0,0 +1,182 @@
+        private Action<NetworkConnection, bool> _handshakeCompletedDelegate;
+        private Action<NetworkConnection, string> _encryptedRequestFromClientDelegate;
+        private Encryptor _crypto;
+
+        #endregion
+
+        /// <summary>
+        /// Use this to send a response to a recently received encrypted message.
+        /// </summary>
+        public void SendResponseToAnEncryptedMessage(NetworkConnection conn, string responseString)
+        {
+            NetworkManager networkManager = InstanceFinder.NetworkManager;
+            networkManager.Log("Sending response to an encrypted request from the client...");
+            ResponseToEncryptedMsgBroadcast responseToEncryptedMsgBroadcast = new()
+            {
+                Response = responseString
+            };
+            networkManager.ServerManager.Broadcast(conn, responseToEncryptedMsgBroadcast, false);
+        }
+
+        private void OnEnable()
+        {
+            InstanceFinder.NetworkManager.ServerManager.OnServerConnectionState += OnServerConnectionState;
+        }
+
+        private void OnDisable()
+        {
+            InstanceFinder.NetworkManager.ServerManager.OnServerConnectionState -= OnServerConnectionState;
+        }
+
+        private void OnServerConnectionState(ServerConnectionStateArgs serverArgs)
+        {
+            NetworkManager networkManager = InstanceFinder.NetworkManager;
+            switch (serverArgs.ConnectionState)
+            {
+                case LocalConnectionState.Started:
+                    _crypto = new Encryptor(EncryptorP, EncryptorG);
+                    //Listen for handshake broadcast from client.
+                    networkManager.ServerManager.RegisterBroadcast<HandshakeRequestBroadcast>(
+                        OnHandshakeRequestBroadcast,
+                        false
+                    );
+                    networkManager.Log("Listening for Handshake request...");
+                    //Listen for EncryptedRequestBroadcast from client.
+                    networkManager.ServerManager.RegisterBroadcast<EncryptedRequestBroadcast>(
+                        OnEncryptedRequestBroadcast,
+                        false
+                    );
+                    networkManager.Log("Listening for encrypted requests...");
+                    break;
+                case LocalConnectionState.Stopped:
+                    networkManager.ServerManager.UnregisterBroadcast<HandshakeRequestBroadcast>(
+                        OnHandshakeRequestBroadcast
+                    );
+                    networkManager.Log("Stopped Listening for Handshake request...");
+                    networkManager.ServerManager.UnregisterBroadcast<EncryptedRequestBroadcast>(
+                        OnEncryptedRequestBroadcast
+                    );
+                    networkManager.Log("Stopped Listening for encrypted requests...");
```

Comment:
is there a reason this line is repeated a second time?
I thought this was related to the default case but i see this is not the case

Reply:
Which line do you mean? Will remove.
Everything here is exactly the same as you had it in OnClientConnectionState(), the only things changing are going from if {} to switch {}, renaming the broadcasts and removing AuthenticationCompleted = false;

Reply:
I cant find it now, strange.
Anyway i will close this comment now 
since there isn't any problem now that
this is comment is about.